### DISCUSSION TITLE
ld: symbol(s) not found for architecture arm64

### DISCUSSION THREAD

**Discussion by florell** [microsoft/vscode-cpptools#10766] [Q&A]
Hello, when I run any program (even "Hello World") I get this error

Undefined symbols for architecture arm64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by Colengms** [comment]
Hi @florell .  The C/C++ Extension provides editor features for C/C++ source files.  It isn't directly involved in compiling and linking code.  What you're describing appears to be an issue with building or linking your code.  The issue may have something to do with the content of your source file(s), and the command line(s) you are using.  If you can provide us more details, such as the contents of the source file(s) you're trying to compile, the command line you're using to build with, or other information about how you're going about it, we might be able to help.

**Comment by kirillmarkelov** [comment]
I got this error when I tried to compile a single file w/o `main` function
```c
#include <stdio.h>

int somename() {
    printf("Hello, World!\n");
    return 0;
}
```

Fixes by having `main` function obviously

**Comment by BruhhIsmaell** [comment]
I kept on getting this error while trying to run the "Helloworld.cpp". I got this fixed by looking into the "tasks.json" file. 

```
{
    "tasks": [
        {
            "type": "cppbuild",
            "label": "C/C++: clang++ build active file",
            "command": "/usr/bin/clang++",
            "args": [
                "-std=c++14",
                "-fcolor-diagnostics",
                "-fansi-escape-codes",
                "-g",
                "${file}",
                "-o",
                "${fileDirname}/${fileBasenameNoExtension}"
            ],
            "options": {
                "cwd": "${fileDirname}"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "detail": "Task generated by Debugger."
        }
    ],
    "version": "2.0.0"
}
```
